### PR TITLE
Fix remaining tests on f4-port

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Summary
Fix remaining tests on f4-port so x86 tests build and pass again.

### Changelist 
- Brake actuation and start switch check on DCM
- Updated GSM vitals msg
- Removed shared error table tests
- Updated json2can submodule (needed to fix some things in the alerts module for shared testing)

### Testing Done

x86 tests pass finally

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: can-bus/json/GSM/GSM_tx.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
-    "GSM_Test1": {
+    "GSM_Vitals": {
```
Comment (will-chaba) [inline can-bus/json/GSM/GSM_tx.json:2]: Need to add Warning message for stackwatermark and tx/rx overflow etc.

File: boards/DCM/Src/App/states/App_InitState.c [hunk 2]
```diff
@@ -24,22 +24,20 @@ static void InitStateRunOnTick100Hz(struct StateMachine *const state_machine)
...
+        const bool was_start_switch_pulled_up = !prev_start_switch_pos && curr_start_switch_pos;
+        prev_start_switch_pos                 = curr_start_switch_pos;
+
+        const bool is_brake_actuated = App_CanRx_FSM_Brake_IsActuated_Get();
 
         // TODO: JSONCAN
         // #ifndef NDEBUG
         //             || App_CanRx_CAN_DEBUGGING_SIGNALS_GetSignal_FORCE_DCM_TO_DRIVE_FROM_INIT(can_rx_interface)
         // #endif
         //         )
```
Comment (will-chaba) [inline boards/DCM/Src/App/states/App_InitState.c:39]: Was this intentionally left?

**Review by will-chaba — CHANGES_REQUESTED** [review]
Looks good, couple of missed items, also in GSM/Src/Io/Io_StackWaterMark.c and GSM/Src/Io/Io_SoftwareWatchdog.c there are still outstanding json2can fixes needed.

**Comment by forbesashli** [comment]
looks good to me

**Review by will-chaba — APPROVED** [review]
:ok_hand:

**Pull request merged**